GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -742,6 +783,17 @@ private void openWithDifftoolToolStripMenuItem_DropDownOpening(object sender, Ev
             selectedParentToLocalToolStripMenuItem.Enabled = _revisionDiffContextMenuController.ShouldShowMenuSelectedParentToLocal(selectionInfo);
             firstParentToLocalToolStripMenuItem.Visible = _revisionDiffContextMenuController.ShouldDisplayMenuFirstParentToLocal(selectionInfo);
             selectedParentToLocalToolStripMenuItem.Visible = _revisionDiffContextMenuController.ShouldDisplayMenuSelectedParentToLocal(selectionInfo);
+
+            var diffFiles = DiffFiles.SelectedItems.ToList();
+            compareTwoSelectedDifftoolToolStripMenuItem.Visible = diffFiles.Count == 2;
+            compareTwoSelectedDifftoolToolStripMenuItem.Enabled = diffFiles.Count == 2
+                                                                  && _revisionDiffContextMenuController.ShouldEnableFirstSpecialCompare(diffFiles?[1])
+                                                                  && _revisionDiffContextMenuController.ShouldEnableSecondSpecialCompare(diffFiles?[0]);
+            compareToExistingDifftoolToolStripMenuItem.Visible = diffFiles.Count == 1 && _savedCompareFileItem != null;
+            compareToExistingDifftoolToolStripMenuItem.Enabled = diffFiles.Count == 1 && _revisionDiffContextMenuController.ShouldEnableSecondSpecialCompare(diffFiles?[0]);
+            compareToExistingDifftoolToolStripMenuItem.Text = _savedCompareFileItem != null ? string.Format(_compareSelectedFile.Text, _savedCompareFileItem.Item.Name) : string.Empty;
+            saveForLaterDifftoolToolStripMenuItem.Visible = diffFiles.Count == 1;
+            saveForLaterDifftoolToolStripMenuItem.Enabled = diffFiles.Count == 1 && _revisionDiffContextMenuController.ShouldEnableFirstSpecialCompare(diffFiles?[0]);
```

Comment:
I like the idea of being able to compare arbitrary files across revisions, but I'm finding the visual presentation of the feature confusing. I had to look at the code to understand the screenshots showed...

Few Qs that immediately popped up:
1. how do I know which files from which revisions I marked?
2. how do I unmark selected/all files?
3. how do I change the order of files?

Reply:
> how do I know which files from which revisions I marked?

The file name is shown in the menu item `Compare selected file to "file"`.
Its revision is something for your short-term memory :)

>    how do I change the order of files?

Click ![grafik](https://user-images.githubusercontent.com/36601201/84541147-17896000-acf7-11ea-92ed-75b4ad66963f.png)
Or replace the selection using `Save selected file for later comparison` and then chose `Compare selected file to "file"` after selecting the first one again.

>    how do I unmark selected/all files?

It's all about comparing exactly two files.
No need to unmark, just start over with `Save selected file for later comparison`.

Reply:
I want to have the previous selectable too
```Diff the selected files``` (when two selected)
```Diff with "file"``` (one selected, revision not included in text)
```Save b/12345678 for later``` (one selected, filename not included)
```Save a/12345678 for later```

Screenshots for other apps. No consistent description. The proposals are considerably longer...
 * No selection
 * Select the same file again 
 * Another selected
 * Two selected
 * more than two selected
![image](https://user-images.githubusercontent.com/6248932/84543230-3b4ea500-acfb-11ea-8213-377eb45216d1.png)

DiffMerge can save several files, as well as clear them
![image](https://user-images.githubusercontent.com/6248932/84547880-b4062f00-ad04-11ea-9512-052998629749.png)

Tortoise supports both git-diff and file system from the context menu
![image](https://user-images.githubusercontent.com/6248932/84547662-4fe36b00-ad04-11ea-8555-a9bda1ccfb40.png)
 
![image](https://user-images.githubusercontent.com/6248932/84543399-96809780-acfb-11ea-8f88-a13782a9c13c.png)

Reply:
Thank you, it's starting to make sense.
Now to pick and agree on the wording :) I think if I saw BC and TG menus I'd conceivably  guess their intent.
Would "Remember file for compare" work? And then "Compare with <the remembered file>"?

Reply:
(I missed this comment, waiting with a merge)
 The menu and Git uses diff rather than compare, so I prefer diff.
"Compare selected..." should be "Diff with <saved file>" (dropping "selected file".

But "Remember Second for later" or "Remember Second" (similar to First) is possible. The shorter, the better is my preference (I constantly fail though.)

@mstv ?